docs(docs-infra): Fix `prefer`/`avoid` rule matching with word bounda…

SkyZeroZx opened the pull request:
Improve code block rule detection by adding word-boundary anchors.
Also update indentation for consistency.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/main/contributing-docs/commit-message-guidelines.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)


## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [x] angular.dev application / infrastructure changes
- [ ] Other... Please describe:


## What is the current behavior?
 
<img width="777" height="913" alt="image" src="https://github.com/user-attachments/assets/daaad9ef-a22f-472b-a5ce-231adb3c7d17" />



## What is the new behavior?
<img width="796" height="882" alt="image" src="https://github.com/user-attachments/assets/5d03e53c-f33e-49e0-a2e4-67e31ac12dfd" />


## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No


<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

JeanMeche reviewed (approved):
Good catch, thanks for that !

SkyZeroZx committed:
docs(docs-infra): Fix prefer/avoid rule matching with word boundaries

Improve code block rule detection by adding word-boundary anchors.
Also update indentation for consistency.
